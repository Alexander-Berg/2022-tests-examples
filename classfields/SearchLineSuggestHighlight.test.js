const React = require('react');
const renderer = require('react-test-renderer');

const SearchLineSuggestHighlight = require('./SearchLineSuggestHighlight');

it('должен правильно подкрасить весь запрос', () => {
    const tree = renderer.create(
        <SearchLineSuggestHighlight
            item={{
                tokens: [
                    { start_char: 0, end_char: 7, type: 'UNRECOGNIZED_FRAGMENT' },
                    { start_char: 22, end_char: 29, type: 'UNRECOGNIZED_FRAGMENT' },
                    { start_char: 31, end_char: 36, type: 'UNRECOGNIZED_FRAGMENT' },
                ],
            }}
            query="странный ауди седан с лазерной пушкой"
        />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
