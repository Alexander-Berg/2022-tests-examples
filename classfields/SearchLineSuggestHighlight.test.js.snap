// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`должен правильно подкрасить весь запрос 1`] = `
<div
  className="SearchLineSuggestHighlight TextInput__control"
>
  <span
    className="SearchLineSuggestHighlight_unparsed"
  >
    странный
  </span>
  <span
    className="SearchLineSuggestHighlight_parsed"
  >
     ауди седан с 
  </span>
  <span
    className="SearchLineSuggestHighlight_unparsed"
  >
    лазерной
  </span>
  <span
    className="SearchLineSuggestHighlight_parsed"
  >
     
  </span>
  <span
    className="SearchLineSuggestHighlight_unparsed"
  >
    пушкой
  </span>
</div>
`;
